net/acme-common/Makefile

```diff
@@ -0,0 +1,100 @@
+#
+# Copyright (C) 2019 Lucian Cristian <lucian.cristian@gmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=acme-common
+PKG_VERSION:=1
+PKG_RELEASE:=1
+
+PKG_MAINTAINER:= \
+    Lucian Cristian <lucian.cristian@gmail.com> \
+    Toke Høiland-Jørgensen <toke@toke.dk>
+PKG_LICENSE:=GPL-3.0-or-later
+PKG_LICENSE_FILES:=COPYING
+
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/acme-common
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=common files for acme and uacme client
+endef
+
+define Package/acme-common/Default/description
+ Common files between acme and uacme app
+endef
+
+define Build/Configure
+endef
+
+define Build/Compile
+endef
+
+define Package/acme-common/conffiles
+/etc/config/acme
+/etc/acme
+endef
+
+define Package/acme-common/install
+	$(INSTALL_DIR) \
+	 $(1)/etc/acme \
+	 $(1)/etc/init.d \
+	 $(1)/etc/config
+
+	$(INSTALL_CONF) ./files/acme.config $(1)/etc/config/acme
+	$(INSTALL_DIR) $(1)/usr/lib/acme
+	$(INSTALL_BIN) ./files/run.sh $(1)/usr/lib/acme/run-acme
+	$(INSTALL_BIN) ./files/acme.init $(1)/etc/init.d/acme
+endef
+
+define Package/luci-app-acme
+  SECTION:=luci
+  CATEGORY:=LuCI
+  TITLE:=ACME package - LuCI interface
+  DEPENDS:= lua luci-base luci-compat +acme-common
+  SUBMENU:=3. Applications
+  PKGARCH:=all
+endef
+
+define Package/luci-app-acme/description
+	Control the ACME Letsencrypt certificate interface
+endef
+
+define Package/luci-app-acme/config
+if PACKAGE_acme-common
+  comment "please also select acme or uacme backend"
```

Comment:
I think we should rather have a dependency on a virtual package ('acme-backend'?), and have both uacme and acme PROVIDE that...

Reply:
both backend can coexist, maybe a check box if both of them are installed ?

Reply:
seems that adding PROVIDES doesn't enforce package selection, maybe is better to split this also ?

Reply:
I'm not sure we want to open the can of worms of supporting both packages simultaneously. Why would anyone want that? :)

Not sure what the best way to implement the "we want one of these" option with opkg... Maybe someone else does? @jow-  ?